components/org.wso2.carbon.identity.api.server.media.service/org.wso2.carbon.identity.api.server.media.service.v1/src/main/resources/media-service.yaml

```diff
@@ -0,0 +1,443 @@
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '501':
+          $ref: '#/components/responses/NotImplemented'
+
+    delete:
+      tags:
+        - Media
+      summary: A user deletes a file.
+      operationId: deleteMyFile
+      security:
+        - BasicAuth: []
+        - OAuth2: []
+      parameters:
+        - $ref: '#/components/parameters/typeParam'
+        - $ref: '#/components/parameters/idParam'
+      responses:
+        '204':
+          description: Resource file and metadata deleted successfully.
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '501':
+          $ref: '#/components/responses/NotImplemented'
+
+  /media/public/{type}/{id}/data:
+    get:
+      tags:
+        - Media
+      summary: Download a publically available file.
+      operationId: downloadPublicFile
+      parameters:
+        - in: query
+          name: identifier
+          schema:
+            type: string
+          example: 'medium'
+        - $ref: '#/components/parameters/typeParam'
+        - $ref: '#/components/parameters/idParam'
+      responses:
+        '200':
+          description: File downloaded successfully.
+          headers:
+            Content-Type:
+              schema:
+                type: string
+            Cache-Control:
+              schema:
+                type: string
+          content:
+            application/octet-stream:
+              schema:
+                $ref: '#/components/schemas/DownloadFile'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '500':
+          $ref: '#/components/responses/ServerError'
+        '501':
+          $ref: '#/components/responses/NotImplemented'
+
+components:
+  parameters:
+    idParam:
+      in: path
+      name: id
+      description: Unique identifier for the file.
+      required: true
+      schema:
+        type: string
+      example: 6e41cb95-c3b3-4e6c-928a-acb1b88e991d
+    typeParam:
+      in: path
+      name: type
+      description: The file type.
+      required: true
+      schema:
+        type: string
+      example: image
+
+  schemas:
+    # Multiple file uploads will not be supported in the first phase of the implementation.
+    ResourceFiles:
+      type: object
+      properties:
+        files:
+          type: array
+          items:
+            type: string
+            format: binary
+        metadata:
+          type: object
+          properties:
+            fileTag:
```

Comment:
shall we remove the file context from the property names? can use tag, security, identifiers in replacement to fileTag, fileSecurity etc..

Reply:
fixed with: https://github.com/wso2/identity-api-server/pull/172/commits/b4eb20175df1c2579ccc6ac562997ef715ab9beb